Framework/HistogramData/src/HistogramItem.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MantidHistogramData/HistogramItem.h"
+#include "MantidHistogramData/Histogram.h"
+#include <algorithm>
+#include <utility>
+
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::Points;
+
+HistogramItem::HistogramItem(const Histogram &histogram, const size_t index)
+    : m_histogram(histogram), m_index(index) {}
+
+double HistogramItem::counts() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Counts) {
+    return histogramRef().counts()[m_index];
+  } else {
+    const Frequencies frequency{histogramRef().frequencies()[m_index]};
+    return Counts(frequency, binEdges())[0];
+  }
+}
+
+double HistogramItem::countVariance() const {
+  return histogramRef().countVariances()[m_index];
+}
+
+double HistogramItem::countStandardDeviation() const {
+  return histogramRef().countStandardDeviations()[m_index];
+}
+
+double HistogramItem::frequency() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Frequencies) {
+    return histogramRef().frequencies()[m_index];
+  } else {
+    const Counts counts{histogramRef().counts()[m_index]};
+    return Frequencies(counts, binEdges())[0];
+  }
+}
+
+double HistogramItem::frequencyVariance() const {
+  return histogramRef().frequencyVariances()[m_index];
+}
+
+double HistogramItem::frequencyStandardDeviation() const {
+  return histogramRef().frequencyStandardDeviations()[m_index];
+}
+
+double HistogramItem::center() const { return point()[0]; }
+
+double HistogramItem::width() const {
+  const auto &edges = binEdges();
+  const auto &lower = edges[0];
+  const auto &upper = edges[1];
+  return upper - lower;
+}
+
+const BinEdges HistogramItem::binEdges() const {
+  const auto &xMode = histogramRef().xMode();
+
+  if (xMode == Histogram::XMode::BinEdges) {
+    const auto &edges = histogramRef().binEdges();
+    const auto &lower = edges[m_index];
+    const auto &upper = edges[m_index + 1];
+    return BinEdges{lower, upper};
+  } else {
+    const auto &points = histogramRef().points();
+    // Guess the bin edges from the points
+    if (points.size() == 0) {
+      // We have no points, just return empty bin edges
+      return BinEdges{};
+    } else if (points.size() == 1) {
+      // We have a single point, let BinEdges guess the edges
+      return BinEdges(points);
+    } else {
+      // We have 2 or more points. Attempt to use up to 4 points
+      // to interpolate the correct bin edges.
```

Comment:
Nice idea, however this is not what what the old conversion from points to bin edges did. `HistogramData` is implemented to follow the old conversion for keeping things consistent. Thus, we have to do the same here.

If you want to do such a change it would need to be done consistently for all cases (legacy implementation, current histogram, and new iterator), but it would definitely need to go by the TSC, and I predict mixed opinions.

In any case, such a change should not be part of this PR.

Reply:
I see that this is different and agree it should not be changed here. I'll make sure it is consistent.